### PR TITLE
Add Ability to author agent evals

### PR DESCRIPTION
**Pull request by JanKrivanek**
### Details

This PR is adding ability to create eval tests for agents - in identical way as they are being authored for skills.
Eval tests for agents explictly calls the tested agent as the entrypoint custom agent for the scenario (so the scenario do not need to craft prompt to invoke it - it is allways present)
It also adds option to specify additional required skills and agents that need to be installed by scenario before running isolated sceanrio

### CODE REVIEW

File: eng/skill-validator/src/Evaluate/EvaluateCommand.cs [hunk 1]
```diff
@@ -1191,25 +1683,128 @@ await File.ReadAllTextAsync(candidate),
...
+        var escapedName = Regex.Escape(targetName.Trim());
         var namePattern = new Regex(
             $@"(?<![\w-]){escapedName}(?![\w-])",
             RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
 
         foreach (var scenario in evalConfig.Scenarios)
         {
             if (namePattern.IsMatch(scenario.Prompt))
-                errors.Add($"Eval scenario '{scenario.Name}' prompt mentions skill name '{skill.Name}' — remove skill name from prompt to avoid biasing baseline runs.");
+                errors.Add($"Eval scenario '{scenario.Name}' prompt mentions skill name '{targetName}' — remove skill name from prompt to avoid biasing baseline runs.");
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/EvaluateCommand.cs]: Fixed. Error message now says 'mentions target name (skill or agent)' - target-neutral wording.

File: eng/skill-validator/src/Evaluate/AgentRunner.cs [hunk 2]
```diff
@@ -374,6 +387,41 @@ internal static SessionConfig BuildSessionConfig(
...
+        if (pluginRoot is not null)
+        {
+            // Plugin run: discover and register all agents in the plugin
+            var pluginAgents = AgentDiscovery.DiscoverAgentsInPlugin(pluginRoot).GetAwaiter().GetResult();
+            if (pluginAgents.Count > 0)
+            {
+                customAgents = pluginAgents.Select(a => BuildCustomAgentConfig(a)).ToList();
+                if (verbose)
+                    log?.Invoke($"      🤖 Registered {customAgents.Count} plugin agent(s): {string.Join(", ", customAgents.Select(a => a.Name))}");
+            }
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/AgentRunner.cs:403]: Fixed. Made BuildSessionConfig async and awaited DiscoverAgentsInPlugin - no more sync-over-async.

File: eng/skill-validator/src/Evaluate/AgentRunner.cs [hunk 3]
```diff
@@ -382,19 +430,49 @@ internal static SessionConfig BuildSessionConfig(
...
                 {
-                    Kind = result ? PermissionRequestResultKind.Approved : PermissionRequestResultKind.DeniedByRules,
+                    Kind = PermissionRequestResultKind.Approved,
                 });
             },
+            Hooks = new SessionHooks
+            {
+                OnPreToolUse = (input, invocation) =>
+                {
+                    var runLabel = skill is not null ? "skilled" : "baseline";
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/AgentRunner.cs]: Fixed. runLabel now checks both agent and pluginRoot to correctly label agent-isolated/agent-plugin runs.

File: eng/skill-validator/src/Evaluate/Judge.cs [hunk 4]
```diff
@@ -41,12 +41,13 @@ public static class Judge
...
             {
-                var result = AgentRunner.CheckPermission(request, options.WorkDir, options.SkillPath, options.Verbose ? log : null, "judge");
+                // Judge sessions: approve all — sandboxing is best-effort
+                // and the judge doesn't execute tool calls in the workspace.
                 return Task.FromResult(new PermissionRequestResult
                 {
-                    Kind = result ? PermissionRequestResultKind.Approved : PermissionRequestResultKind.DeniedByRules,
+                    Kind = PermissionRequestResultKind.Approved,
                 });
             },
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/Judge.cs:52]: Fixed. Changed to DeniedByRules - judges should not execute tools.

File: eng/skill-validator/README.md [hunk 5]
```diff
@@ -254,6 +258,9 @@ scenarios:
 |--------|-------------|
 | `copy_test_files` | When `true`, copies all files from the eval directory (except `eval.yaml`) into the agent working directory before each run. Useful when test fixtures live alongside the eval file. |
 | `files` | Explicit list of files to create in the working directory. Each entry has a `path` and either inline `content` or a `source` path (relative to the skill directory). Applied after `copy_test_files`. |
+| `commands` | Shell commands to run in the work directory before the agent starts (e.g. `dotnet build -bl:build.binlog`). |
+| `additional_required_skills` | List of skill names (from the same plugin) to load in the **isolated** run alongside the target. Useful when an agent routes to specific skills or a skill depends on sibling skills. Does not affect baseline (nothing loaded) or plugin (everything loaded) runs. |
+| `additional_required_agents` | List of agent names (from the same plugin) to register in the **isolated** run alongside the target. Same semantics as `additional_required_skills` but for agents. |
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/README.md:263]: Fixed. Added additional_required_skills/agents resolution to the skill eval ExecuteRun path too, so docs are now accurate for both skill and agent evals.

File: eng/skill-validator/src/Evaluate/Models.cs [hunk 6]
```diff
@@ -56,7 +56,9 @@ public sealed record SetupFile(
 public sealed record SetupConfig(
     bool CopyTestFiles = false,
     IReadOnlyList<SetupFile>? Files = null,
-    IReadOnlyList<string>? Commands = null);
+    IReadOnlyList<string>? Commands = null,
+    IReadOnlyList<string>? AdditionalRequiredSkills = null,
+    IReadOnlyList<string>? AdditionalRequiredAgents = null);
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/Models.cs:61]: Fixed. Skill eval now resolves and passes these deps to the isolated run (same as agent eval). Both paths are symmetric now.

File: eng/skill-validator/src/Evaluate/EvaluateCommand.cs [hunk 7]
```diff
@@ -1191,25 +1683,128 @@ await File.ReadAllTextAsync(candidate),
...
+        foreach (var name in skillNames)
+        {
+            var match = allSkills.FirstOrDefault(s =>
+                s.Name.Equals(name, StringComparison.OrdinalIgnoreCase)
+                || Path.GetFileName(s.Path).Equals(name, StringComparison.OrdinalIgnoreCase));
+            if (match is not null)
+                resolved.Add(match);
+            else
+                Console.Error.WriteLine($"{Ansi.Yellow}⚠  additional_required_skills: '{name}' not found in plugin at '{pluginRoot}'{Ansi.Reset}");
+        }
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/EvaluateCommand.cs:1790]: Fixed. Changed from warning+continue to throw InvalidOperationException - unresolved required dependencies now fail fast.

File: eng/skill-validator/src/Evaluate/AgentRunner.cs [hunk 8]
```diff
@@ -118,28 +120,32 @@ public static Task CleanupWorkDirs(bool keepSessions = false)
...
-            else if (data.TryGetValue("fileName", out var fileVal) && fileVal is JsonElement fileEl && fileEl.ValueKind == JsonValueKind.String)
-                reqPath = fileEl.GetString() ?? "";
-            else if (data.TryGetValue("fullCommandText", out var cmdVal) && cmdVal is JsonElement cmdEl && cmdEl.ValueKind == JsonValueKind.String)
-                reqPath = cmdEl.GetString() ?? "";
+            if (args.TryGetProperty(key, out var val) && val.ValueKind == JsonValueKind.String)
+                return val.GetString();
         }
 
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/AgentRunner.cs:139]: Fixed. Added 8 focused unit tests for ExtractPathFromToolArgs covering all edge cases.

File: eng/skill-validator/src/Evaluate/PairwiseJudge.cs [hunk 9]
```diff
@@ -80,13 +80,13 @@ public static class PairwiseJudge
...
-                var extraDirs = options.SkilledWorkDir is not null ? new[] { options.SkilledWorkDir } : null;
-                var result = AgentRunner.CheckPermission(request, options.WorkDir, options.SkillPath, options.Verbose ? log : null, "pairwise-judge", additionalAllowedDirs: extraDirs);
+                // Pairwise judge sessions: approve all — sandboxing is best-effort
+                // and the judge doesn't execute tool calls in the workspace.
                 return Task.FromResult(new PermissionRequestResult
                 {
-                    Kind = result ? PermissionRequestResultKind.Approved : PermissionRequestResultKind.DeniedByRules,
+                    Kind = PermissionRequestResultKind.Approved,
                 });
             },
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/PairwiseJudge.cs:91]: Fixed. Changed to DeniedByRules - pairwise judge sessions should not execute tools.

File: .agents/skills/create-skill-test/SKILL.md [hunk 10]
```diff
@@ -129,6 +140,24 @@ setup:
...
+  additional_required_skills:
+    - binlog-failure-analysis    # loaded in isolated run alongside the target
+  additional_required_agents:
+    - build-perf                 # registered in isolated run alongside the target
+```
+
+- Names are resolved from the same plugin's `skills/` or `agents/` directory.
+- These only affect the **isolated** run. The **plugin** run already loads everything; the **baseline** loads nothing.
+- Different scenarios of the same target can declare different dependencies (per-scenario granularity).
+- If a declared name cannot be resolved, the validator fails with an error.
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply .agents/skills/create-skill-test/SKILL.md:159]: Implementation now throws InvalidOperationException on unresolved deps, matching what the doc states. No doc change needed.

File: eng/skill-validator/src/Evaluate/EvaluateCommand.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/EvaluateCommand.cs]: Fixed. Error message now says mentions target name (skill or agent) - target-neutral wording.

[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/EvaluateCommand.cs]: Fixed. Error message now says mentions target name (skill or agent) - target-neutral wording.

File: eng/skill-validator/src/Evaluate/AgentRunner.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/AgentRunner.cs:403]: Fixed. Made BuildSessionConfig async and awaited DiscoverAgentsInPlugin - no more sync-over-async.

File: eng/skill-validator/src/Evaluate/AgentRunner.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/AgentRunner.cs]: Fixed. runLabel now checks both agent and pluginRoot to correctly label agent-isolated/agent-plugin runs.

File: eng/skill-validator/src/Evaluate/EvaluateCommand.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/EvaluateCommand.cs:1790]: Fixed. Changed to throw InvalidOperationException - unresolved deps now fail fast.

File: eng/skill-validator/src/Evaluate/AgentRunner.cs (hunk 8 above)
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/AgentRunner.cs:139]: Fixed. Added 8 focused unit tests for ExtractPathFromToolArgs.

File: .agents/skills/create-skill-test/SKILL.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply .agents/skills/create-skill-test/SKILL.md:159]: Implementation now throws InvalidOperationException on unresolved deps. Doc already correct.

File: eng/skill-validator/src/Evaluate/Models.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/Models.cs:61]: Fixed. Skill eval now resolves and passes these deps to the isolated run. Both skill and agent paths are symmetric now.

File: eng/skill-validator/src/Evaluate/PairwiseJudge.cs (hunk 9 above)
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/skill-validator/src/Evaluate/PairwiseJudge.cs:91]: Fixed. Changed to DeniedByRules - pairwise judge sessions should not execute tools.
